feat: Switch schedule_viewer.py map from Google to Leaflet

losvedir opened the pull request:
Since the change to the Google Maps API that now requires an API key with billing enabled, we can't easily use this. Leaflet with OSM is a great alternative, and the Leaflet JS API is fairly similar to Google Maps, so it was fairly straightforward to switch to it. 

It should be nearly functionally identical. There was a bit of behavior in the code around selected stop text that I didn't keep since Leaflet doesn't have a `marker.getText()` and `marker.setText()` equivalent. I could potentially put that in a `Popup` but then I'd have to bind Popups to all the markers. I never used this functionality before, and with Google Maps broken now I couldn't test it out, so I'm not really sure what we're missing. The page works in all the ways I've used it.

To try it out, it should be as easy as cloning this repo and checking out this branch, and running `python2.7 schedule_viewer.py --feed=/path/to/build/current/`.

<img width="1256" alt="Screen Shot 2021-04-09 at 3 37 07 PM" src="https://user-images.githubusercontent.com/384428/114238373-1e4e2700-994a-11eb-849f-fec204ff39c1.png">

losvedir committed:
feat: Switch schedule_viewer.py map from Google to Leaflet

paulswartz commented on gtfsscheduleviewer/files/index.js:
```diff
@@ -11,25 +10,18 @@ function load(config) {
-  map = new google.maps.Map(map_dom, mapOptions);
+
+  map = L.map('map')
+  L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
```
Can you use our tiles? `https://cdn.mbta.com/osm_tiles/{z}/{x}/{y}.png` (attribution can stay the same).

paulswartz reviewed (approved):
one small note, otherwise 🍰

jfabi reviewed (commented):
>There was a bit of behavior in the code around selected stop text that I didn't keep since Leaflet doesn't have a `marker.getText()` and `marker.setText()` equivalent. I could potentially put that in a Popup but then I'd have to bind Popups to all the markers. I never used this functionality before, and with Google Maps broken now I couldn't test it out, so I'm not really sure what we're missing.

I don't know if this is the exact behavior you're referring to, but there used to be these labels next to each stop marker showing the arrival/departure time for the respective stop. I think they're pretty handy because sometimes the popups won't actually show the _selected_ trip's time if there are too many trips around the time you select in the UI.

It looks like you could probably use [tooltips in Leaflet (which bind to markers)](https://leafletjs.com/reference-1.7.1.html#tooltip) to recreate the old behavior, or you could add the time to the existing markers (maybe in a bigger/bolder type at the beginning of the body?).

(Screenshots from previous pull requests.)

![image](https://user-images.githubusercontent.com/3793006/114406540-9fd5cd00-9b75-11eb-8978-45e28983a5b6.png)
![image](https://user-images.githubusercontent.com/3793006/114406839-dad80080-9b75-11eb-90c8-2321b1cd10ff.png)

A more minor qualm, really with the default OSM style, is that a lot more roads blend in with bus routes with little color contrast.

<img width="649" alt="Screenshot 2021-04-09 at 18 16 44" src="https://user-images.githubusercontent.com/3793006/114412151-d4985300-9b7a-11eb-8bf9-6761035fdd87.png">
<img width="828" alt="Screenshot 2021-04-09 at 18 14 53" src="https://user-images.githubusercontent.com/3793006/114412173-da8e3400-9b7a-11eb-9aa7-2a32edb310b6.png">

I tried using another basemap style that's free/doesn't require a key, and it seemed maybe marginally better, but I'm not sure if it's enough to warrant a change—see below. Let me know if you'd prefer it as a likely future user, though, and I can get you the details.

<img width="540" alt="Screenshot 2021-04-12 at 11 24 34" src="https://user-images.githubusercontent.com/3793006/114422354-314c3b80-9b84-11eb-8b53-0aecdd853e8a.png">
<img width="754" alt="Screenshot 2021-04-12 at 11 21 02" src="https://user-images.githubusercontent.com/3793006/114422358-327d6880-9b84-11eb-8190-4ef12b986ae4.png">

losvedir commented:
@jfabi - I updated the tiles to use the ones we actually serve, and it clashes less. I also updated to use permanent tool-tips to display the times. Let me know what you think.

(I amended the commit rather than pushing up a fixup, so you'll have to `fetch` and then `reset --hard origin/gjd-maps-to-leaflet`.)

<img width="899" alt="Screen Shot 2021-04-12 at 1 11 49 PM" src="https://user-images.githubusercontent.com/384428/114441449-05838280-9b91-11eb-9f8e-26aa3ad024e2.png">

paulswartz reviewed (approved):
🍰

jfabi reviewed (approved):
Excellent work! 🎆
